ui/cypress/integration/compatibility-features/user_login_spec.js

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 PingCAP, Inc. Licensed under Apache-2.0.
+
+// Test User Login Compatibility
+// FEATURE_VERSION < 5.3.0 of TiDB Dashboard does not support nonRootLogin
+// FEATURE_VERSION >= 5.3.0 of TiDB Dashboard supports nonRootLogin
+describe('User Login', () => {
+  if (Cypress.env('FEATURE_VERSION') === '6.0.0') {
+    // Create user test
+    before(() => {
+      let query = "DROP USER IF EXISTS 'test'@'%'"
+      let password = ''
+      cy.task('queryDB', { query, password })
+
+      query = "CREATE USER 'test'@'%' IDENTIFIED BY 'test_pwd'"
+      cy.task('queryDB', { query, password })
+
+      query = "GRANT ALL PRIVILEGES ON *.* TO 'test'@'%' WITH GRANT OPTION"
+      cy.task('queryDB', { query, password })
+    })
+
+    // Run before each test
+    beforeEach(function () {
+      // Load a fixed set of data located in cypress/fixtures.
+      cy.fixture('uri.json').as('uri')
+
+      // Direct to login page
+      cy.visit('@uri.root')
+    })
+
+    it('noRootLogin is supported', () => {
+      cy.log('FEATURE_VERSION is: ', Cypress.env('FEATURE_VERSION'))
+      cy.request(`${Cypress.env('apiUrl')}info/info`).then((resp) => {
+        expect(resp.status).to.eq(200)
+        expect(resp.body.supported_features).include('nonRootLogin')
```

Comment:
Do we really need to test the api response? From the user's side, maybe BDD is enough. If the business process doesn't change, the test is valid. But the implementation may change.

Reply:
I agree that from the user side, this check may not be necessary, but here we want to check whether the TiDB Dashboard with `FEATURE_VERSION >= 5.3.0` support `nonRootLogin` or not.  This is a real API test instead of mocking data to assert the supported_features.